File: org.kevoree.modeling.microframework/src/main/java/org/kevoree/modeling/api/Manager.java
package org.kevoree.modeling.api;

import org.kevoree.modeling.api.events.ModelElementListener;

/**
 * Created by duke on 9/30/14.
 */

public interface Manager {

    public Dimension create();

    public Dimension get(String key);

    public void saveAll(Callback<Boolean> callback);

    public void deleteAll(Callback<Boolean> callback);

    public void unloadAll(Callback<Boolean> callback);


    public void listen(ModelElementListener listener, Long from, Long to, String path);

    public void disable(ModelElementListener listener);

    public void stream(String query, Callback<KObject> callback);

}

File: org.kevoree.modeling.microframework/src/main/java/org/kevoree/modeling/api/Manager.kt
package org.kevoree.modeling.api

import org.kevoree.modeling.api.events.ModelElementListener

/**
 * Created by duke on 9/30/14.
 */

public trait Manager {

    fun create(): Dimension
    fun get(key: String)
    fun saveAll(callback: Callback<Boolean>)
    fun deleteAll(callback: Callback<Boolean>)
    fun unloadAll(callback: Callback<Boolean>)


    fun listen(listener: ModelElementListener, from: Long?, to: Long?, path: String)
    fun disable(listener: ModelElementListener)
    fun stream(query: String, callback: Callback<KObject<*>>)

}